includes/class-access-manager.php

```diff
@@ -0,0 +1,235 @@
+<?php
+/**
+ * Newspack Print Integration Access Manager.
+ *
+ * @package Newspack\PrintCirculationIntegration
+ */
+
+namespace Newspack\PrintCirculationIntegration;
+
+use WP_Error;
+
+/**
+ * Class to manage access and memberships for imported users.
+ */
+class Access_Manager {
+
+	/**
+	 * Get configured membership plans.
+	 *
+	 * @return array Array of membership plan IDs.
+	 */
+	public static function get_membership_plans() {
+		return Settings::get_setting( Settings::DEFAULT_MEMBERSHIPS_OPTION, [] );
+	}
+
+	/**
+	 * Grant membership access to a user.
+	 *
+	 * @param int    $user_id User ID.
+	 * @param array  $plan_ids Optional. Specific plan IDs to grant. Default empty array (uses configured plans).
+	 * @param string $source Optional. Source of the membership. Default 'newspack-print-circ-import'.
+	 * @return bool|WP_Error True on success, False on no plans granted, WP_Error on failure.
+	 */
+	public static function grant_membership_access( $user_id, $plan_ids = [], $source = 'newspack-print-circ-import' ) {
+		if ( ! class_exists( 'WC_Memberships_User_Membership' ) ) {
+			return new WP_Error(
+				'missing_memberships_class',
+				__( 'WooCommerce Memberships is not active.', 'newspack-print' )
+			);
+		}
+
+		/**
+		 * Set default plans to grant.
+		 */
+		$default_plans_to_grant = ! empty( $plan_ids ) ? $plan_ids : self::get_membership_plans();
+		$default_plans_to_grant = array_map( 'absint', $default_plans_to_grant );
+		$user_status            = Newspack_Fields::get_field_value( Newspack_Fields::STATUS, $user_id );
+
+		if ( empty( $default_plans_to_grant ) ) {
+			return false;
+		}
+
+		/**
+		 * Check for which plans to update and which to grant.
+		 */
+		$existing_memberships = self::get_user_memberships( $user_id );
+
+		// Log existing memberships.
+		if ( ! empty( $existing_memberships ) ) {
+			Logger::add_log(
+				sprintf(
+					'User %d already has active memberships: %s',
+					$user_id,
+					implode( ', ', $existing_memberships )
+				)
+			);
+		}
+
+		/**
+		 * Next we update the status of any existing memberships that are also in the default plans to grant.
+		 * This is to ensure that if a user already has a membership, we can update its status.
+		 * This is useful for cases where the user has a membership but its status changes.
+		 */
+		$plans_to_update = array_intersect( $default_plans_to_grant, $existing_memberships );
+
+		// Update existing memberships.
+		if ( ! empty( $plans_to_update ) ) {
+			foreach ( $plans_to_update as $plan_id ) {
+				$membership = wc_memberships_get_user_membership( $user_id, $plan_id );
+	
+				if ( ! is_wp_error( $membership ) ) {
+					// Update status.
+					$membership->update_status( $user_status );
```

Comment:
at this point, let's also add an order note (`$user_membership->add_note(...`) to the membership saying that its status got update via a CSV import.